libqtile/core/manager.py

```diff
@@ -408,14 +408,78 @@ def update_net_desktops(self) -> None:
         self.core.update_net_desktops(self.groups, index)
 
     def update_client_list(self) -> None:
-        """Updates the client stack list
+        """Update X11 client list
 
         This is needed for third party tasklists and drag and drop of tabs in
         chrome
         """
         windows = [wid for wid, c in self.windows_map.items() if c.group]
         self.core.update_client_list(windows)
 
+    def update_client_order(self) -> None:
+        """Update X11 client order
+        This is needed for third party tasklists and drag and drop of tabs in
+        chrome
+        """
+        windows = [wid for wid, c in self.windows_map.items() if c.group]
+        stack = self.conn.query_tree()
+        # TODO: there appear to be some windows (starting VirtualBox creates one) that might not be properly unmanaged
+        windows = sorted([w for w in windows if w in stack], key=lambda wid: stack.index(wid))
+        self.core.update_client_order(windows)
+
+    def change_layer(self, wid, up=True):
+        """Raise a window above its peers or move it below them, depending on 'up'.
+        Raising a normal window will not lift it above pinned windows etc.
+        """
+        client = self.windows_map.get(wid, None)
+        if client is None:
+            return
+
+        above = client.window.net_wm_state_above
+        below = client.window.net_wm_state_below
+
+        def _is_peer(wid):
+            # TODO: We could add more flags to check to adhere to EWMH-standards:
+            # https://specifications.freedesktop.org/wm-spec/1.3/ar01s07.html#STACKINGORDER
+            wnd = self.windows_map[wid].window
+            if above:
+                return wnd.net_wm_state_above
+            elif below:
+                return wnd.net_wm_state_below
+            else:
+                return not wnd.net_wm_state_above and not wnd.net_wm_state_below
+
+        stack = [w for w in self.conn.query_tree() if w in self.windows_map]
+        if len(stack) == 1:
```

Comment:
Should this also return if `len(stack)` equals 0?

Reply:
It should, but since we should have at least one window (which this method is being called from), an exception might be interesting to notice.

Reply:
I reworked the function completely, now this comparison is a `< 2`.